### PR TITLE
Safety fixes

### PR DESCRIPTION
**Pull request by Manishearth**
This code is currently unsound as it initializes an `&[u8]` that points to potential padding bytes.

https://github.com/rust-vmm/vhost/blob/6ca88e160a8d34be54a158cc1bc702e9250e97dd/crates/vhost/src/vhost_user/master.rs#L202

`repr(C)` would also work though it is brittle if the struct is change in the future, happy to switch over.

I also marked `VhostUserMsgValidator` as requiring `ByteValued` since `extract_request_body` assumes these types are POD.

https://github.com/rust-vmm/vhost/blob/6ca88e160a8d34be54a158cc1bc702e9250e97dd/crates/vhost/src/vhost_user/slave_req_handler.rs#L728

### CODE REVIEW

File: crates/vhost/src/backend.rs [hunk 1]
```diff
@@ -65,6 +65,7 @@ impl VringConfigData {
 
 /// Memory region configuration data.
 #[derive(Default, Clone, Copy)]
+#[repr(packed)]
```
Comment (sboeuf) [inline crates/vhost/src/backend.rs:68]: The commit message doesn't match the code.
And please add a body to your commit message to explain why this change is needed.

File: crates/vhost/src/vhost_user/message.rs [hunk 2]
```diff
@@ -203,7 +203,7 @@ impl Req for SlaveReq {
 }
 
 /// Vhost message Validator.
-pub trait VhostUserMsgValidator {
+pub trait VhostUserMsgValidator: ByteValued {
```
Comment (sboeuf) [inline crates/vhost/src/vhost_user/message.rs:206]: Please explain why this is needed in the commit message.

**Review by sboeuf — APPROVED** [review]

**Review by stefano-garzarella — APPROVED** [review]

**Comment by stefano-garzarella** [comment]
@Manishearth thanks for the PR, merged!
